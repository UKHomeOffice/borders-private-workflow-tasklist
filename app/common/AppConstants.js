
export default class AppConstants {
    static DASHBOARD_PATH = "/dashboard";
    static YOUR_TASKS_PATH = "/your-tasks";
    static YOUR_GROUP_TASKS_PATH = '/your-group-tasks';
    static PROCEDURES_PATH = '/procedures';
    static REPORTS_PATH = '/reports';
    static REPORT_PATH = "/report";
    static MESSAGES_PATH = "/messages";
    static CALENDAR_PATH = "/calendar";
    static START_A_PROCEDURE = "/start-a-procedure";
    static TASK_PATH="/task";
    static ADMIN_PATH="/admin";
    static SHIFT_PATH = "/shift";
    static PROCEDURE_DIAGRAM_PATH = "/procedure-diagram";
    static ONBOARD_USER_PATH="/onboard-user";
    static MANDATORY_DECLARATION_PATH="/mandatory-declarations";
    static  ONE_MINUTE = 60000;

}
