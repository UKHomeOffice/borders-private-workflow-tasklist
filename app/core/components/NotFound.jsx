import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { list } from 'react-immutable-proptypes';
import { errors as errorsSelector } from '../error/selectors';
import AppConstants from '../../common/AppConstants';

export class NotFoundPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      is404: null,
    };
  }

  componentDidMount() {
    this.checkErrors();
  }

  componentDidUpdate(prevProps) {
    const { errors } = this.props;
    if (errors !== prevProps.errors) {
      this.checkErrors();
    }
  }

  checkErrors() {
    const { errors } = this.props;
    const is404 = errors.some(error => error.get('status') === 404);
    document.title = `${is404 ? 'Page not found' : 'Error'} | ${
      AppConstants.APP_NAME
    }`;
    this.setState({
      is404,
    });
  }

  render() {
    const { resource, id } = this.props;
    const { is404 } = this.state;
    switch (is404) {
      case true:
        return (
          <Fragment>
            <h2 className="govuk-heading-l govuk-!-margin-top-6">
              Page not found
            </h2>
            <p className="govuk-body">
              If you typed the web address, check it is correct.
            </p>
            <p className="govuk-body">
              If you pasted the web address, check you copied the entire
              address.
            </p>
            <p className="govuk-body">
              If you&rsquo;re looking for a form then{' '}
              <a href="/forms">view the full available list here</a>.
            </p>
            <p className="govuk-body">
              To find something else then{' '}
              <a href="/dashboard">go to the COP dashboard</a>.
            </p>
            <p className="govuk-body">
              If the web address is correct or you selected a link or button,{' '}
              <a
                href="https://support.cop.homeoffice.gov.uk/servicedesk/customer/portal/3"
                rel="noopener noreferrer"
                target="_blank"
              >
                contact the COP support team
              </a>{' '}
              to report a bug.
            </p>
          </Fragment>
        );

      case false:
        return (
          <div>
            <h4 className="govuk-heading-s">
              {resource} with identifier {id} was not found
            </h4>
          </div>
        );

      default:
        return null;
    }
  }
}
NotFoundPage.propTypes = {
  errors: list.isRequired,
  resource: PropTypes.string.isRequired,
  id: PropTypes.string,
};

NotFoundPage.defaultProps = {
  id: '',
};

export default connect(state => ({
  errors: errorsSelector(state),
}))(NotFoundPage);
