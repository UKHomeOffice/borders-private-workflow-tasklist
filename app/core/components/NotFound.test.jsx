import React from 'react';
import configureStore from 'redux-mock-store';
import { Map, List } from 'immutable';
import AppConstants from '../../common/AppConstants';
import NotFound, { NotFoundPage } from './NotFound';

describe('NotFound page', () => {
  const props = {
    errors: List([
      Map({
        status: 500,
        message: 'test',
      }),
    ]),
    resource: 'Form',
  };

  it('renders without crashing', () => {
    const wrapper = mount(<NotFoundPage {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('renders standard notice if not 404 error', () => {
    const wrapper = shallow(<NotFoundPage {...props} />);
    expect(wrapper.find('h4').text()).toBe(
      'Form with identifier  was not found',
    );
    expect(global.window.document.title).toBe(
      `Error | ${AppConstants.APP_NAME}`,
    );
  });

  it('renders 404 page if 404 error', () => {
    const wrapper = shallow(<NotFoundPage {...props} />);
    wrapper.setProps({
      ...props,
      errors: List([
        Map({
          status: 404,
          message: 'test',
        }),
      ]),
    });
    expect(wrapper.find('h2').text()).toBe('Page not found');
    expect(global.window.document.title).toBe(
      `Page not found | ${AppConstants.APP_NAME}`,
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<NotFoundPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('NotFound container', () => {
  it('receives state as expected', () => {
    const store = configureStore()({
      'error-page': Map({
        errors: List([
          Map({
            status: 404,
            message: 'test',
          }),
        ]),
      }),
    });
    const wrapper = shallow(<NotFound store={store} resource="Form" />);
    expect(wrapper.exists()).toBe(true);
  });
});
