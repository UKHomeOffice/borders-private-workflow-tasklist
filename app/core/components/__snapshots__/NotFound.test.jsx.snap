// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page matches snapshot 1`] = `
<div>
  <h4
    className="govuk-heading-s"
  >
    Form
     with identifier 
     was not found
  </h4>
</div>
`;
